File: app/src/main/java/com/example/core/localization/AppLocalizations.kt
package com.example.core.localization

object AppLocalizations {

    fun getString(key: String, isArabic: Boolean = false): String {
        val enMap = mapOf(
            "app_name" to "PFMS Finance",
            "net_worth" to "Net Worth",
            "accounts" to "Accounts",
            "transactions" to "Transactions",
            "assets" to "Assets",
            "debts" to "Debt Center",
            "budgets" to "Budgets",
            "goals" to "Goals",
            "bills" to "Bills & Subscriptions",
            "analytics" to "Analytics & Reports",
            "quick_actions" to "Quick Actions",
            "add_income" to "Add Income",
            "add_expense" to "Add Expense",
            "transfer" to "Transfer",
            "new_asset" to "New Asset",
            "new_debt" to "New Debt",
            "new_goal" to "New Goal",
            "pay_bill" to "Pay Bill",
            "reports" to "Reports",
            "recent_activities" to "Recent Activities",
            "receivables" to "Receivables (Owed to You)",
            "payables" to "Payables (You Owe)",
            "settings" to "Settings & Security",
            "security" to "Security & Lock",
            "cloud_backup" to "Cloud Sync & Backup",
            "currency" to "Base Currency",
            "language" to "Language / اللغة",
            "statement_pdf" to "Export PDF Statement"
        )

        val arMap = mapOf(
            "app_name" to "النظام المالي الشخصي",
            "net_worth" to "صافي الثروة",
            "accounts" to "الحسابات المالية",
            "transactions" to "المعاملات المالية",
            "assets" to "الأصول والممتلكات",
            "debts" to "مركز الديون",
            "budgets" to "الميزانيات",
            "goals" to "الأهداف المالية",
            "bills" to "الفواتير والاشتراكات",
            "analytics" to "التحليلات والتقارير",
            "quick_actions" to "الإجراءات السريعة",
            "add_income" to "إضافة دخل",
            "add_expense" to "إضافة مصروف",
            "transfer" to "تحويل مال",
            "new_asset" to "أصل جديد",
            "new_debt" to "دين جديد",
            "new_goal" to "هدف جديد",
            "pay_bill" to "دفع فاتورة",
            "reports" to "التقارير المالية",
            "recent_activities" to "آخر العمليات",
            "receivables" to "ديون لك (مستحقات)",
            "payables" to "ديون عليك (التزامات)",
            "settings" to "الإعدادات والأمان",
            "security" to "الأمان والحماية",
            "cloud_backup" to "المزامنة السحابية",
            "currency" to "العملة الأساسية",
            "language" to "اللغة / Language",
            "statement_pdf" to "تصدير كشف حساب PDF"
        )

        return if (isArabic) arMap[key] ?: key else enMap[key] ?: key
    }
}

File: lib/core/localization/app_localizations.dart
class AppLocalizations {
  static String getString(String key, {bool isArabic = false}) {
    const enMap = {
      "app_name": "PFMS Finance",
      "net_worth": "Net Worth",
      "accounts": "Accounts",
      "transactions": "Transactions",
      "assets": "Assets",
      "debts": "Debt Center",
      "budgets": "Budgets",
      "goals": "Goals",
      "bills": "Bills & Subscriptions",
      "analytics": "Analytics & Reports",
      "quick_actions": "Quick Actions",
      "add_income": "Add Income",
      "add_expense": "Add Expense",
      "transfer": "Transfer",
      "new_asset": "New Asset",
      "new_debt": "New Debt",
      "new_goal": "New Goal",
      "pay_bill": "Pay Bill",
      "reports": "Reports",
      "recent_activities": "Recent Activities",
      "receivables": "Receivables (Owed to You)",
      "payables": "Payables (You Owe)",
      "settings": "Settings & Security",
      "security": "Security & Lock",
      "cloud_backup": "Cloud Sync & Backup",
      "currency": "Base Currency",
      "language": "Language / اللغة",
      "statement_pdf": "Export PDF Statement"
    };

    const arMap = {
      "app_name": "النظام المالي الشخصي",
      "net_worth": "صافي الثروة",
      "accounts": "الحسابات المالية",
      "transactions": "المعاملات المالية",
      "assets": "الأصول والممتلكات",
      "debts": "مركز الديون",
      "budgets": "الميزانيات",
      "goals": "الأهداف المالية",
      "bills": "الفواتير والاشتراكات",
      "analytics": "التحليلات والتقارير",
      "quick_actions": "الإجراءات السريعة",
      "add_income": "إضافة دخل",
      "add_expense": "إضافة مصروف",
      "transfer": "تحويل مال",
      "new_asset": "أصل جديد",
      "new_debt": "دين جديد",
      "new_goal": "هدف جديد",
      "pay_bill": "دفع فاتورة",
      "reports": "التقارير المالية",
      "recent_activities": "آخر العمليات",
      "receivables": "ديون لك (مستحقات)",
      "payables": "ديون عليك (التزامات)",
      "settings": "الإعدادات والأمان",
      "security": "الأمان والحماية",
      "cloud_backup": "المزامنة السحابية",
      "currency": "العملة الأساسية",
      "language": "اللغة / Language",
      "statement_pdf": "تصدير كشف حساب PDF"
    };

    if (isArabic) {
      return arMap[key] ?? key;
    } else {
      return enMap[key] ?? key;
    }
  }
}
